src/ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp

```diff
@@ -670,6 +672,49 @@ void compute_dest_vars_from_source_vars(
         dest_vars, source_vars, mesh, jac_grid_to_inertial,
         invjac_grid_to_inertial, jac_logical_to_grid, invjac_logical_to_grid,
         inertial_mesh_velocity);
+  } else if constexpr (any_index_in_frame_v<SourceTags, Frame::Inertial> and
+                       any_index_in_frame_v<typename InterpolationTargetTag::
+                                                vars_to_interpolate_to_target,
+                                            Frame::Distorted>) {
+    // Need to do frame transformations to Distorted frame.
+    const auto& functions_of_time = get<domain::Tags::FunctionsOfTime>(cache);
+    const auto& block = domain.blocks().at(element_id.block_id());
+    ASSERT(block.has_distorted_frame(),
+           "Cannot interpolate to distorted frame in a block that does not "
+           "have a distorted frame");
+    ElementMap<3, ::Frame::Grid> map_element_logical_to_grid{
+        element_id, block.moving_mesh_logical_to_grid_map().get_clone()};
+    domain::CoordinateMaps::Composition element_logical_to_distorted_map{
+        domain::element_to_block_logical_map(element_id),
+        block.moving_mesh_logical_to_grid_map().get_clone(),
+        block.moving_mesh_grid_to_distorted_map().get_clone()};
+    const auto [inertial_coords, invjac_distorted_to_inertial,
+                jac_distorted_to_inertial,
+                distorted_to_inertial_mesh_velocity] =
+        block.moving_mesh_distorted_to_inertial_map()
+            .coords_frame_velocity_jacobians(
+                element_logical_to_distorted_map(
+                    logical_coordinates(mesh),
+                    InterpolationTarget_detail::
+                        evaluate_temporal_id_for_expiration(temporal_id),
+                    functions_of_time),
+                InterpolationTarget_detail::evaluate_temporal_id_for_expiration(
```

Comment:
Why is this not `InterpolationTarget_detail::get_temporal_id_value`? I know it's used above, but looking at the two functions, they are identical for all types except `TimeStepId`.  The difference is that `get_temporal_id` does `TimeStepId.substep_time().value()` while `evaluate_temporal_id_for_expiration` does `TimeStepId.step_time().value()`. They are different, but does that matter here?

Reply:
That's a good question.  `evaluate_temporal_id_for_expiration` is used in the code for the other frames too, and I thought that was what @wthrowe wanted. @wthrowe should I be using `get_temporal_id_value` here (and for the other frames) instead?

Reply:
@wthrowe all the tests pass if I put those `ASSERT`s into both functions.   Should `evaluate_temporal_id_for_expiration` be removed, and only `get_temporal_id_value` used everywhere?  (and should I keep the ASSERT?)

Reply:
Both of these were added in #3421. It seems that `evaluate_temporal_id_for_expiration` was added because previously, `VerifyTemporalIdsAndSendPoints.hpp` used `pending_id.step_time().value()`. Whereas, `get_temporal_id_value` was added because `ObserveTimeSeriesOnSurface.hpp` used `temporal_id.substep_time().value()`.

So we needed one that got the `step_time` and the other that got the `substep_time`. However, I'm not sure if we need to use one or the other in each circumstance. I agree with @wthrowe that `_value` seems to just get the current time, but unclear what `_expiration` is actually needed for.

Reply:
Looking more closely, I don't think any of the uses of `step_time()` here are correct.  (Uses of `step_time()` outside time-related code are usually wrong.)  So just get rid of that function and probably rename the other one to get rid of the now-unnecessary disambiguator.

Reply:
Ok, I will keep the one using `substep_time()` and remove the other one.  Should I leave the ASSERT in there?

Reply:
I say yes. It's just an `ASSERT` so won't affect productions stuff

Reply:
No.  I intend to add valid cases where it will fail.